src/subdomains/core/route/route.controller.ts

```diff
@@ -37,6 +40,13 @@ export class RouteController {
     ]).then(([buy, sell, swap]) => ({ buy, sell, swap, crypto: swap }));
   }
 
+  @Get('payment/:id')
+  @ApiExcludeEndpoint()
+  async getPaymentRoute(@Param('id') id: string): Promise<SellDto> {
+    const sellRoute = await this.sellService.getPaymentRoute(id);
+    return this.sellController.toDto(sellRoute);
+  }
```

Comment:
I think I would place this endpoint in `PaymentLinkController` under `GET paymentLink/recipient`.

Maybe also with query parameters instead of path parameter? I'm not sure 🤔

Reply:
Maybe we can then somehow share the checks performed on the route? They are already in `PaymentLinkService`.